providers/update.rb

```diff
@@ -0,0 +1,88 @@
+#
+# Author:: Baptiste Courtois (<b.courtois@criteo.com>)
+# Cookbook Name:: wsus-client
+# Provider:: update
+#
+# Copyright:: Copyright (c) 2015 Criteo.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# API documentation: http://msdn.microsoft.com/en-us/library/windows/desktop/aa387099.aspx
+
+use_inline_resources
+
+def whyrun_supported?
+  true
+end
+
+action :download do
+  updates_to_download = updates.reject(&:IsDownloaded)
+  return if updates_to_download.count == 0
+
+  converge_by "downloading #{updates_to_download.count} update(s)" do
+    # Transforms to a new update collection
+    update_collection = WIN32OLE.new('Microsoft.Update.UpdateColl')
```

Comment:
You are right I'll fix it right now! It currently works because the windows cookbook is loading the library for me :)